pkg/types/alibabacloud/platform.go

```diff
@@ -7,11 +7,32 @@ type Platform struct {
 
 	// ResourceGroupID is the ID of an already existing resource group where the cluster should be installed.
 	// This resource group must be empty with no other resources when trying to use it for creating a cluster.
-	// If empty, a new resource group will created for the cluster.
-	// Destroying the cluster using installer will delete this resource group.
+	// If empty, a new resource group will created for the cluster,
+	// and destroying the cluster using installer will delete this resource group.
 	// +optional
 	ResourceGroupID string `json:"resourceGroupID,omitempty"`
 
+	// VpcID is the ID of an already existing VPC where the cluster should be installed.
+	// If empty, a new VPC will created for the cluster, and destroying
+	// the cluster using installer will delete this VPC.
+	// +optional
+	VpcID string `json:"vpcID,omitempty"`
+
+	// VSwitchIDs is the ID list of already existing VSwitches where cluster resources will be created.
```

Comment:
I assume that since the vSwitch is part of the VPC, that the vSwitches can only be user-provided if the VPC is also user-provided. If so, then there should (1) be text to that affect in this description, (2) be validation that the VPC ID is specified when the vSwitch IDs are specified, and (3) be validation that the vSwitches are part of the VPC.

Reply:
Regarding (2) and (3), relevant  validation  have already been made, and I will add  text  in this description

> that the vSwitches can only be user-provided if the VPC is also user-provided.

I disagree with this point. I think it’s okay for the user to only provide the VPC ID, and the installer will create the VSwitches

Reply:
Sorry, I thought that I had deleted parts (2) and (3) after re-checking that the validation was indeed already there.

> > that the vSwitches can only be user-provided if the VPC is also user-provided.
> 
> I disagree with this point. I think it’s okay for the user to only provide the VPC ID, and the installer will create the VSwitches

Yes, I agree that it is OK to use a user-provided VPC and installer-generated VSwitches. My point is the converse. If the user provides the VSwitches, then they *must* all provide the VPC. I would like to call that out specifically in the description for the field so that the user does not assume that the installer will deduce the VPC to use from the VSwitches (which the installer could presumably do).